lightningd/Makefile

```diff
@@ -39,6 +39,7 @@ LIGHTNINGD_LIB_SRC :=				\
 	lightningd/channel.c			\
 	lightningd/channel_config.c		\
 	lightningd/commit_tx.c			\
+	lightningd/connection.c			\
```

Comment:
I called it connection since initially I wanted to move the queue management stuff for peer-connections in here as well. But with the custom loops that are tightly coupled with the state machines, that's easier said than done :smiley: 

I'll do the `msg_queue` merge, should not be hard.